src/ConstructiveSolidGeometry/plotting/CSG/CSG.jl

```diff
@@ -6,33 +6,44 @@
     )
 end
 
-@recipe function f(csg::AbstractConstructiveGeometry{T}, primitive::AbstractPrimitive{T}, axis::Symbol, slice::T, st::Symbol; n_samples = missing, CSG_scale = missing) where {T}
+@recipe function f(csg::AbstractConstructiveGeometry{T}, primitive::AbstractPrimitive{T}, axis::Symbol, slice::T, st::Symbol; n_samples = 40, CSG_scale = missing, projection = :none, full_det = false, linewidth = :auto) where {T}
     if st == :samplesurface
         CSG_scale = ismissing(CSG_scale) ? get_scale(csg) : CSG_scale
         spacing = T(CSG_scale/n_samples)
         filter(p -> in(p,csg), sample(primitive, spacing))
     elseif st == :slice
+        isgr = occursin("GRBackend", string(typeof(plotattributes[:plot_object].backend)))
         CSG_scale = ismissing(CSG_scale) ? get_scale(csg) : CSG_scale
         pointtype, axisrot = CartesianPoint, axis
         spacing = T(CSG_scale/n_samples)
         if axis == :φ
             pointtype, axisrot, slice, spacing = CylindricalPoint, :y, 0, spacing*T(0.5)
         end
-        samples = filter(pt -> abs(getproperty(pt, axisrot) - slice) < spacing/2, filter(p -> in(p,csg), sample(primitive, spacing)))
+        samples = filter(pt -> in(pt, csg) && abs(getproperty(pt, axisrot) - slice) < spacing/2, sample(primitive, spacing))
         proj = fieldnames(CartesianPoint)[findall(x -> x != axis, fieldnames(pointtype))]
         u = getproperty.(samples, proj[1])
         v = getproperty.(samples, proj[2])
-        if axis == :φ
+        if axis == :φ && !full_det
             idx = findall(0 .≤ u)
             u = u[idx]
             v = v[idx]
         end
         proj = filter(x -> x != axis, fieldnames(pointtype))
-        xunit --> internal_length_unit
         yunit --> internal_length_unit
         xguide --> string(proj[1])
         yguide --> string(proj[2])
-        internal_length_unit*u, internal_length_unit*v
+        if linewidth != :auto
+            markersize := linewidth
+        end
```

Comment:
Didn't try this out, but does this also work with `lw`?

I would naively assume, that one would need something like `linewidth = seriesattribute[:linewidth]` in case the user uses an alias.

Reply:
Yes, `lw` and `proj` are supported by having `linewidth` and `projection` in the keywords of the plot recipe.